crates/wdl-lint/RULES.md

```diff
@@ -1,53 +0,0 @@
-# Rules
```

Comment:
I don't view this as a good change, as it's the only place where there is a plaintext description of each of the rules that are available, what their tags are, and what a brief description is. I understand that it sometimes gets out of sync, but another solution is just to ensure we have a check to make sure this table is updated in the PR checklist.

For me, I don't think we should delete this until we have a better strategy around publishing the available rules and associated information in the docs.

Reply:
And I didn't say this explicitly, but I don't think asking non-Rust-familiar users to dig through our developer documentation to find out this information is a good enough solution in the meantime.